eng/SourceBuild.props

```diff
@@ -0,0 +1,69 @@
+<Project>
+
+  <PropertyGroup>
+    <GitHubRepositoryName>aspnetcore</GitHubRepositoryName>
```

Comment:
Would this file ever be used in a local (non-CI) build❔ If not, probably better to use AzDO's `$(Build.Repository.Name)` predefined variable which is available as `$(Build_Repository_Name)` w/in MSBuild because most predefined variables are put into the environment.

Reply:
Yes this gets used during local builds.  The source-build infrastructure uses this name when building the source-build intermediate NuGet package name.

Reply:
We should used the predefined variable and fallback to this value if it's not set.

Reply:
@JunTaoLuo, Can you explain the value in depending on `$(Build_Repository_Name)` since it won't exist on the dev scenario.  Doesn't having the same code path in the local dev flow vs CI flow have value?  Having a different value or at least source of the value seems like a potential source of breakage.

Reply:
Fair enough, let's keep this as is.